### PR TITLE
Add `--frozen` flag support for `bundle package`

### LINKED ISSUE #3356: "bundle package" support for --frozen
Trying to use "bundle package --frozen", but it's not a supported switch as it is with "bundle install".

```
$ bundle package --frozen
Unknown switches '--frozen'
```

Internally "bundle package" calls "bundle install", which it passes its parsed options to, so it should be trivial to support this and the entire "bundle install" options should probably be supported by "bundle package".

### PR DESCRIPTION
**Pull request by RochesterinNYC**
- Closes #3356

### CODE REVIEW

File: spec/commands/package_spec.rb [hunk 1]
```diff
@@ -99,6 +99,32 @@
...
+
+    subject { bundle "package --frozen" }
+
+    it "tries to install with frozen" do
+      gemfile <<-G
+        source "file://#{gem_repo1}"
+        gem "rack"
+        gem "rack-obama"
+      G
+      subject
```
Comment (segiddins) [inline spec/commands/package_spec.rb]: This is supposed to fail, right? If so, `expect_err` and test the exitstatus?

Comment (RochesterinNYC) [inline spec/commands/package_spec.rb]: Yup, thanks for the catch.

**Comment by RochesterinNYC** [comment]
PR is updated.

**Comment by segiddins** [comment]
:+1: 
@indirect r?

**Comment by indirect** [comment]
Awesome, thanks!

@homu r+

**Pull request merged**